src/View/ViewBuilder.php

```diff
@@ -310,27 +310,39 @@ public function getPlugin(): ?string
      */
     public function addHelper(string $helper, array $options = [])
     {
-        if ($options) {
-            $array = [$helper => $options];
-        } else {
-            $array = [$helper];
+        $this->_helpers[$helper] = $options;
+
+        return $this;
+    }
+
+    /**
+     * Adds helpers to use by merging with existing ones.
+     *
+     * @param array $helpers Helpers to use.
+     * @return $this
+     * @since 5.0.0
+     */
+    public function addHelpers(array $helpers)
+    {
+        foreach ($helpers as $helper => $config) {
+            if (is_int($helper)) {
+                $helper = $config;
+                $config = [];
+            }
+            $this->addHelper($helper, $config);
         }
 
-        return $this->setHelpers($array);
+        return $this;
     }
 
     /**
      * Sets the helpers to use.
      *
      * @param array $helpers Helpers to use.
-     * @param bool $merge Whether or not to merge existing data with the new data.
      * @return $this
      */
-    public function setHelpers(array $helpers, bool $merge = true)
+    public function setHelpers(array $helpers)
     {
-        if ($merge) {
```

Comment:
We should deprecate this flag in 4.3

Reply:
https://github.com/cakephp/cakephp/pull/15898